modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointMarkersStorage.java

```diff
@@ -65,28 +88,41 @@
     /** File I/O factory for writing checkpoint markers. */
     private final FileIOFactory ioFactory;
 
+    /** Checkpoint read-write lock. */
+    private final CheckpointReadWriteLock lock;
+
     /** Checkpoint metadata directory ("cp"), contains files with checkpoint start and end */
     public final File cpDir;
 
     /** Temporary write buffer. */
     private final ByteBuffer tmpWriteBuf;
 
+    /** Counter representing quantity of checkpoints since last checkpoint history snapshot. */
+    private final AtomicInteger checkpointSnapshotCounter = new AtomicInteger(1);
```

Comment:
I think it's better **AtomicLong** or use **AtomicInteger#updateAndGet** with reset value

Reply:
Why `long` if `earliestCpChangesThreshold` is `int`?